crates/types/src/message.rs

```diff
@@ -34,9 +33,6 @@ use std::{fmt::Debug, marker::PhantomData};
 #[derive(Serialize, Deserialize, Clone, Debug, Derivative, PartialEq, Eq, Hash)]
 #[serde(bound(deserialize = "", serialize = ""))]
 pub struct Message<TYPES: NodeType> {
```

Comment:
Just checking for my own understanding: `NodeType` is versioned right? (So `Message` is implicitly versioned through that?)

Reply:
Not right now, no. I'm working on a way to do that...

Reply:
Sorry, a bit of clarification. Nothing implements the trait `Versioned`, which has been added in `vbs`; it was meant to be added alongside Serialize and Deserialize as a feature macro, e.g.

```rust
#[derive(Serialize, Deserialize, Versioned)]
#[base_version(1, 3)]
#[max_version(1, 7)]
struct MyStruct {
}
```

but I've come to realize that we usually don't really know the actual version in crates that have generic types...

We may end up needing a crate-level version and a mechanism for mapping the versions of crates to the actual version, but that's not trivial. Another solution might be a tagged, but never versioned, repository containing a single "version" crate that gets bound in the executable target's `Cargo.lock`, and injects versions into our library crates.

Reply:
`Message` is not inherently bound to a specific protocol version, though a change to `Message` will force versioning once `Versioned` is implemented. Not by way of `NodeType`, though.

I'm still trying to come up with a nice way to inject versions into the consensus by way of something coupled to `NodeType`, but giving it generic parameters is catastrophically contagious.